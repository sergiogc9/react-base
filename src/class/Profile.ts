export interface ProfileObject {}

export class Profile {

	// public user: User;

	// constructor(profile: ProfileObject) {
	// 	this.user = new User(profile.user);
	// }
}
