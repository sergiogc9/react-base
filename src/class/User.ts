export class UserObject {
	public id: string;
	public name: string;

	constructor(values?: Partial<UserObject>) {
		if (values) Object.assign(this, values);
	}
}

export class User extends UserObject {

}
