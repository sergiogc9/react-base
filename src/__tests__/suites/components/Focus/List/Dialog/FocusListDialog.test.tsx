import React from 'react';
import { mount } from 'enzyme';

import FocusListDialog from '@src/components/Focus/List/Dialog/FocusListDialog';
import { FeedObject } from '@src/class/Feed';
import { FocusObject, FocusFeeds } from '@src/class/Focus';
import TestHelper from '@src/__tests__/utils/Helper/TestHelper';

const document = TestHelper.getDocument();

const feed: FeedObject = {
	definition: {},
	deleted_at: new Date(1549287550563),
	enabled: true,
	filters: {},
	focus_id: "focus-id-1",
	id: "feed-id-1",
	inserted_at: new Date(1549287550563),
	name: "feed-name-1",
	oldest_document: document,
	type: "online",
	updated_at: new Date(1549287550563)
};

const feed2: FeedObject = {
	...feed,
	id: "feed-id-2",
	name: "feed-name-2",
	type: "socialmedia",
};

const focusFeeds: FocusFeeds = {
	online: [feed],
	socialmedia: [feed2],
	print: []
};

const focus: FocusObject = {
	deleted_at: new Date(1549287550563),
	feeds: focusFeeds,
	id: "focus-id-1",
	inserted_at: new Date(1549287550563),
	name: "focus-name-1",
	oldest_document: document,
	updated_at: new Date(1549287550563),
	url_logo: "http://fake-url.com"
};

let onCloseMock: jest.Mock;
let onCreateMock: jest.Mock;
let onRenameMock: jest.Mock;
let onRemoveMock: jest.Mock;
let onRemoveFeedMock: jest.Mock;

describe('<FocusListDialog />', () => {
	let wrapper: any;

	beforeEach(() => {
		onCloseMock = jest.fn();
		onCreateMock = jest.fn();
		onRenameMock = jest.fn();
		onRemoveMock = jest.fn();
		onRemoveFeedMock = jest.fn();
		wrapper = mount((<FocusListDialog
			action="createFocus"
			focus={focus}
			feed={null}
			onClose={onCloseMock}
			onCreateFocus={onCreateMock}
			onRemoveFocus={onRemoveMock}
			onRenameFocus={onRenameMock}
			onRemoveFeed={onRemoveFeedMock}
		></FocusListDialog>));
		jest.useFakeTimers();
	});

	it("Test FocusListDialog snapshot creating a focus", () => {
		wrapper = mount((<FocusListDialog
			action="createFocus"
			focus={focus}
			feed={null}
			onClose={onCloseMock}
			onCreateFocus={onCreateMock}
			onRemoveFocus={onRemoveMock}
			onRenameFocus={onRenameMock}
			onRemoveFeed={onRemoveFeedMock}
		></FocusListDialog>));
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test FocusListDialog snapshot renaming a focus", () => {
		wrapper = mount((<FocusListDialog
			action="renameFocus"
			focus={focus}
			feed={null}
			onClose={onCloseMock}
			onCreateFocus={onCreateMock}
			onRemoveFocus={onRemoveMock}
			onRenameFocus={onRenameMock}
			onRemoveFeed={onRemoveFeedMock}
		></FocusListDialog>));
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test FocusListDialog snapshot removing a focus", () => {
		wrapper = mount((<FocusListDialog
			action="removeFocus"
			focus={focus}
			feed={null}
			onClose={onCloseMock}
			onCreateFocus={onCreateMock}
			onRemoveFocus={onRemoveMock}
			onRenameFocus={onRenameMock}
			onRemoveFeed={onRemoveFeedMock}
		></FocusListDialog>));
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test FocusListDialog snapshot removing a feed", () => {
		wrapper = mount((<FocusListDialog
			action="removeFocus"
			focus={focus}
			feed={feed}
			onClose={onCloseMock}
			onCreateFocus={onCreateMock}
			onRemoveFocus={onRemoveMock}
			onRenameFocus={onRenameMock}
			onRemoveFeed={onRemoveFeedMock}
		></FocusListDialog>));
		expect(wrapper.html()).toMatchSnapshot();
	});

	it("Test FocusListDialog should show error", () => {
		expect(wrapper.find('#focusListSimpleDialog .md-text--error')).toHaveLength(0);
		wrapper.setState({ dialogInputError: "fake_error" });
		expect(wrapper.find('#focusListSimpleDialog .md-text--error')).toHaveLength(2);
	});

	it("Test FocusListDialog should call onClose", () => {
		wrapper.find('Button#launchmetricsDialogCancelBtn').simulate('click');
		expect(onCloseMock).toHaveBeenCalledTimes(1);
	});

	it("Test FocusListDialog should call onDialogAccepted in each action case", () => {
		expect(onCreateMock).toHaveBeenCalledTimes(0);
		expect(onRenameMock).toHaveBeenCalledTimes(0);
		expect(onRemoveMock).toHaveBeenCalledTimes(0);
		wrapper.find('input#focusListCreateFocusInput').simulate('change', { target: { value: 'Focus name' } });
		wrapper.find('Button#launchmetricsDialogAcceptBtn').simulate('click');
		expect(onCreateMock).toHaveBeenCalledTimes(1);
		wrapper.setProps({ action: "renameFocus" });
		wrapper.find('input#focusListRenameFocusInput').simulate('change', { target: { value: 'Focus name' } });
		wrapper.find('Button#launchmetricsDialogAcceptBtn').simulate('click');
		expect(onRenameMock).toHaveBeenCalledTimes(1);
		wrapper.setProps({ action: "removeFocus" });
		wrapper.find('Button#launchmetricsDialogAcceptBtn').simulate('click');
		expect(onRemoveMock).toHaveBeenCalledTimes(1);
		wrapper.setProps({ action: "removeFeed" });
		wrapper.find('Button#launchmetricsDialogAcceptBtn').simulate('click');
		expect(onRemoveFeedMock).toHaveBeenCalledTimes(1);
	});

	it("Test FocusListDialog should call onDialogAccepted with error (empty input)", () => {
		expect(wrapper.state().dialogInputError).toBe(null);
		wrapper.find('Button#launchmetricsDialogAcceptBtn').simulate('click');
		expect(wrapper.state().dialogInputError).not.toBe(null);
		wrapper.setProps({ action: "renameFocus" });
		wrapper.find('Button#launchmetricsDialogAcceptBtn').simulate('click');
		expect(wrapper.state().dialogInputError).not.toBe(null);
	});
});
