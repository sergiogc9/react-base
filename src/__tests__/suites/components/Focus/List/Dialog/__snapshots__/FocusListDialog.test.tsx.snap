// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FocusListDialog /> Test FocusListDialog snapshot creating a focus 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusListSimpleDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content md-dialog-content--padded\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">focus.prompt_new_focus</div></div><div class=\\"dialog-content\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input\\"><label for=\\"focusListCreateFocusInput\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">name</label><input id=\\"focusListCreateFocusInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div><div class=\\"dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span>"`;

exports[`<FocusListDialog /> Test FocusListDialog snapshot removing a feed 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusListSimpleDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content md-dialog-content--padded\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">focus.remove.confirm</div></div><div class=\\"dialog-content\\"><div>focus.remove.notice</div></div><div class=\\"dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span>"`;

exports[`<FocusListDialog /> Test FocusListDialog snapshot removing a focus 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusListSimpleDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content md-dialog-content--padded\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">focus.remove.confirm</div></div><div class=\\"dialog-content\\"><div>focus.remove.notice</div></div><div class=\\"dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span>"`;

exports[`<FocusListDialog /> Test FocusListDialog snapshot renaming a focus 1`] = `"<span class=\\"md-dialog-container md-overlay md-overlay--active md-pointer--hover launchmetrics-simple-dialog\\" tabindex=\\"-1\\"><div aria-labelledby=\\"launchmetrics-dialog\\" id=\\"focusListSimpleDialog\\" role=\\"dialog\\" class=\\"md-paper md-paper--5 md-dialog md-dialog--centered md-background--card\\"><section class=\\"md-dialog-content md-dialog-content--padded\\"><span aria-hidden=\\"true\\"></span><div class=\\"dialog-container\\"><div class=\\"dialog-header md-grid\\"><div class=\\"dialog-text\\">focus.rename.title</div></div><div class=\\"dialog-content\\"><div class=\\"md-text-field-container md-full-width md-text-field-container--input\\"><label for=\\"focusListRenameFocusInput\\" class=\\"md-floating-label md-floating-label--inactive md-floating-label--inactive-sized md-text--secondary\\">name</label><input id=\\"focusListRenameFocusInput\\" type=\\"text\\" class=\\"md-text-field md-text-field--floating-margin md-full-width md-text\\" value=\\"focus-name-1\\"><hr class=\\"md-divider md-divider--text-field md-divider--expand-from-left\\"></div></div><div class=\\"dialog-footer md-grid \\"><button id=\\"launchmetricsDialogCancelBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--grey btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.cancel</button><button id=\\"launchmetricsDialogAcceptBtn\\" type=\\"button\\" class=\\"md-btn md-btn--flat md-btn--text md-pointer--hover md-text md-inline-block btn--green btn--no-background\\"><div class=\\"md-ink-container\\"></div>form.accept</button></div></div></section></div></span>"`;
